src/test/java/net/snowflake/ingest/SimpleIngestIT.java

```diff
@@ -212,4 +229,61 @@ private void getHistoryAndAssertLoad(SimpleIngestManager manager, String test_fi
       assertTrue(loaded);
     }
   }
+
+  @Test
+  public void testHeadersForIngestFileRequest() throws Exception {
+    TestUtils.executeQuery("put file://" + testFilePath + " @" + stageName);
+
+    // create ingest manager
+    SimpleIngestManager manager = TestUtils.getManager(pipeName);
+
+    // create a file wrapper
+    StagedFileWrapper myFile = new StagedFileWrapper(TEST_FILE_NAME, null);
+    RequestBuilder requestBuilder = manager.getRequestBuilder();
+    final String additionUserAgentInfo = "kafka-source/NONE";
+    HttpPost postWithAdditionUserAgentInfo =
+        requestBuilder.generateInsertRequest(
+            UUID.randomUUID(),
+            pipeName,
+            Collections.singletonList(myFile),
+            false,
+            additionUserAgentInfo);
+    verifyDefaultUserAgent(
+        postWithAdditionUserAgentInfo.getAllHeaders(), true, additionUserAgentInfo);
+
+    // Passing null and empty string would also work
+    HttpPost nullAdditionalUserAgent =
+        requestBuilder.generateInsertRequest(
+            UUID.randomUUID(), pipeName, Collections.singletonList(myFile), false, null);
+    verifyDefaultUserAgent(nullAdditionalUserAgent.getAllHeaders(), false, null);
+    HttpPost emptyAdditionalUserAgent =
+        requestBuilder.generateInsertRequest(
+            UUID.randomUUID(), pipeName, Collections.singletonList(myFile), false, null);
+    verifyDefaultUserAgent(emptyAdditionalUserAgent.getAllHeaders(), false, null);
+  }
+
+  private void verifyDefaultUserAgent(
+      final Header[] headers,
+      final boolean verifyAdditionalUserAgentInfo,
+      final String httpUserAgentInformation) {
+    for (Header h : headers) {
+      if (h.getName().equalsIgnoreCase(HttpHeaders.USER_AGENT)) {
+        if (verifyAdditionalUserAgentInfo) {
+          assertTrue(h.getValue().contains(httpUserAgentInformation));
```

Comment:
Maybe we can assert that it endsWith the custom agent info?

Reply:
Will do..